DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/task_sequences_parallel_loops/README.md

```diff
@@ -0,0 +1,370 @@
+  for (size_t i = 0; i < len; i++) {
+    int tmp = PipeAB::read();
+    tmp += i;
+    PipeBC::write(tmp);  
+  }
+}
+
+[[intel::use_stall_enable_clusters]] 
+void LoopC(int len) {
+  [[intel::initiation_interval(1)]]  
+  for (size_t i = 0; i < len; i++) {
+    int tmp = PipeBC::read();
+    tmp += i;
+    PipeCD::write(tmp);  
+  }
+}
+
+[[intel::use_stall_enable_clusters]] 
+void LoopD(int len) {
+  [[intel::initiation_interval(1)]]  
+  for (size_t i = 0; i < len; i++) {
+    int tmp0 = PipeCD::read();
+    int tmp1 = PipeAD::read();
+    int out = tmp0 + tmp1;
+    PipeOut::write(out);  
+  }
+}
+
+///////////////////////////////////////
+
+struct OptimizedKernel {
+  int len;
+
+  void operator()() const {
+    sycl::ext::intel::experimental::task_sequence<LoopA> task_a;
+    sycl::ext::intel::experimental::task_sequence<LoopB> task_b;
+    sycl::ext::intel::experimental::task_sequence<LoopC> task_c;
+    sycl::ext::intel::experimental::task_sequence<LoopD> task_d;
+
+    task_a.async(len);
+    task_b.async(len);
+    task_c.async(len);
+    task_d.async(len);
+  }
+};
+```
+The C++ functions `LoopA`, `LoopB`, `LoopC` and `LoopD` are the task functions with asynchronous activities defined. Each of the task functions is used to parameterize a specific `task_sequence` class to create four `task_sequence` object instances, `task_a`, `task_b`,`task_c` and `task_d`. The task functions are called in parallel through `async()` method that accepts the same arguments as their respective task functions. The
+`async()` method call is non-blocking, and it returns before the asynchronous task function finishes executing, and potentially before the task function even _begins_ executing, as the return type of the `async()` method provides no implicit information on the execution status of task functions.
+
+You can see a concrete example of a kernel that naively uses four sequential loops in `naive/main.cpp`. This design is modified to use task sequences to run the loops concurrently in `task_sequence/main.cpp`.
+
+## Sample Structure
+The 2 different example designs in this sample perform similar operations. You may compare the C++ source files to see the code changes that are necessary to convert a design to use task sequences instead of sequential loops.
+
+1. [Naive](naive/main.cpp) This implementation contains sequential loops. Arrays are used to pass data between the loops.
+2. [Task sequence](task_sequence/main.cpp) This implementation uses task sequences to schedule loops to run concurrently. Pipes are used to pass data between the loops.
+
+## Build a Design
+>**Note**: When working with the command-line interface (CLI), you should configure the oneAPI toolkits using environment variables. Set up your CLI environment by sourcing the `setvars` script in the root of your oneAPI installation every time you open a new terminal window. This practice ensures that your compiler, libraries, and tools are ready for development.
+>
+> Linux*:
```

Comment:
As oneAPI 2024.0, there are two possible directory layout and two ways to set environment variables:
`setvars` and `oneapi-vars`

https://www.intel.com/content/www/us/en/docs/oneapi/programming-guide/2024-1/use-the-setvars-and-oneapi-vars-scripts-with-linux.html

https://www.intel.com/content/www/us/en/docs/oneapi/programming-guide/2024-1/use-the-setvars-and-oneapi-vars-scripts-with.html


Do your instructions need to account for both scripts here?